pdk/src/main/java/com/pinterest/android/pdk/PDKClient.java

```diff
@@ -527,7 +527,7 @@ private void saveAccessToken(String accessToken) {
         SharedPreferences sharedPref = _context.getSharedPreferences(PDK_SHARED_PREF_FILE_KEY, Context.MODE_PRIVATE);
         SharedPreferences.Editor editor = sharedPref.edit();
         editor.putString(PDK_SHARED_PREF_TOKEN_KEY, accessToken);
-        editor.commit();
+        editor.apply();
```

Comment:
It will be better to use `commit()` on a separate thread, apart from the UI thread. The reason is `commit()` is synchronous while `apply()` is asynchronous. So in case it might not perform actions immediately as expected.

Reply:
Should I create a separate thread for using commit?

Reply:
Yes, or it would be better to create a generic class for the same.

Reply:
How should I achieve this?

Reply:
Create a Java class that takes an object and call a function to execute the same on a separate thread.